python/pyspark/ml/util.py

```diff
@@ -464,7 +464,10 @@ def saveMetadata(
         metadataJson = DefaultParamsWriter._get_metadata_to_save(
             instance, sc, extraMetadata, paramMap
         )
-        sc.parallelize([metadataJson], 1).saveAsTextFile(metadataPath)
+        spark = SparkSession.getActiveSession()
+        spark.createDataFrame([(metadataJson,)], schema=["value"]).repartition(1).write.text(
```

Comment:
I just checked that pyspark classic (which is built atop `LogicalRDD`) need repartition to make sure single partition;
while pyspark connect doesn't need this, since it uses `LocalRelation`.

Probably we can use `coalesce(1)` instead.